x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -108,7 +108,11 @@ public Analyzer(Configuration configuration, FunctionRegistry functionRegistry,
                 new ResolveAggsInHaving()
                 //new ImplicitCasting()
                 );
-        return Arrays.asList(substitution, resolution);
+        Batch finish = new Batch("Finish Analysis",
+                new PruneSubqueryAliases(),
```

Comment:
Why don't we merge them in one rule?

Reply:
Because they have different concerns - I would expect the second one to change when we add full support for subselects.